### PR TITLE
Implement authCheck

### PR DESCRIPTION
**Pull request by WilliamChan4**
Implements route protection using a check Authenticated middleware function.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #### Custom ####
 
-*.env
+**.env
```
Comment (WilliamChan4) [inline .gitignore:3]: This was from pulling master

File: src/backend/src/routes/index.ts [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
 import UserRouter from "./Users";
 import ChallengeRouter from "./Challenges";
-import { UNAUTHORIZED } from "http-status-codes";
+import { checkAuthenticated } from 'src/controllers/auth';
 
 // Init router and path
 const router = Router();
 
-// user logged in 
-export const authCheck = (req: Request, res: Response, next: () => void) => {
```
Comment (WilliamChan4) [inline src/backend/src/routes/index.ts:10]: For some reason, the authCheck function needed to be in a different module.

**Comment by WilliamChan4** [comment]
@soroushMosaddad99 I've (manually) checked that all the routes work. Obviously it would be better if we had some unit tests, but I think I've spent enough time fixing the backend 😄

**Comment by soroushMosaddad99** [comment]
thank you William, good job!

**Pull request merged**